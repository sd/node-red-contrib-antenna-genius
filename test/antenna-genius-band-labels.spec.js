const helper = require("node-red-node-test-helper");
const bandLabelsNode = require("../antenna-genius-band-labels.js");
const serverNode = require("../antenna-genius-server.js");

const nodes = [bandLabelsNode, serverNode];

describe('antenna-genius-band-labels Node', () => {

  const flow = [
    { id: "n1", type: "antenna-genius-server", name: "test name", hostname: "localhost", port: 9007, disabledColor: "Black", activeColor: "Green", selectedColor: "Blue", autoConnect: false},
    { id: "n2", type: "antenna-genius-band-labels", name: "test name", server: "n1", wires:[["n3"]]},
    { id: "n3", type: "helper" }
  ];

  beforeEach(() => {
  });

  afterEach(() => {
    helper.unload();
  });

  it('should be loaded', done => {
    helper.load(nodes, flow, () => {
        let n2 = helper.getNode("n2");
        expect(n2).toBeDefined();
        done();
      });
  });

  it('should have properties', done => {
    helper.load(nodes, flow, () => {
        let n2 = helper.getNode("n2");
        expect(n2).toHaveProperty('name', 'test name');
        done();
      });
  });


  it('should provide no output with no input', done => {
    helper.load(nodes, flow, () => {
        let n2 = helper.getNode("n2");
        n2.server.updatesEventEmitter.emit("status", false);
        expect(n2.send.notCalled).toBeTruthy();
        done();
      });
  });

  it('should provide no output with no band definitions', done => {
    helper.load(nodes, flow, () => {
        let n2 = helper.getNode("n2");
        n2.server.status = { portA_band: 0, portB_band: 1 };
        n2.server.updatesEventEmitter.emit("status", false);
        expect(n2.send.notCalled).toBeTruthy();
        done();
      });
  });

  it('should provide output', done => {
    helper.load(nodes, flow, () => {
        let n2 = helper.getNode("n2");
        n2.server.status = { portA_band: 0, portB_band: 1 };
        n2.server.bands = [{ band_name: "bandA" }, { band_name: "bandB" }];
        n2.server.info = { name: "Name" };

        let n3 = helper.getNode("n3");
        n3.on("input", (msg) => {
            try {
              expect(msg).toHaveProperty("payload", { bandLabelA: "bandA", bandLabelB: "bandB" });
              expect(n2.send.calledOnce).toBeTruthy();
              expect(n2.status.calledOnceWithExactly({ fill: "green", shape: "dot", text: "Name - bandA/bandB" })).toBeTruthy();
              done();
            } catch (err) {
              done(err);
            }
          });

        n2.server.updatesEventEmitter.emit("status", false);
      });
  });

  it('should not provide connected status', done => {
    helper.load(nodes, flow, () => {
        let n2 = helper.getNode("n2");
        n2.server.updatesEventEmitter.emit("connected");
        expect(n2.status.notCalled).toBeTruthy();
        done();
      });
  });

  it('should provide connected status', done => {
    helper.load(nodes, flow, () => {
        let n2 = helper.getNode("n2");
        n2.server.info = { name: "Name" };
        n2.server.updatesEventEmitter.emit("connected");
        expect(n2.status.calledOnceWithExactly({ fill: "green", shape: "dot", text: "Name" })).toBeTruthy();
        done();
      });
  });

  it('should provide disconnected status', done => {
    helper.load(nodes, flow, () => {
        let n2 = helper.getNode("n2");
        n2.server.info = { name: "Name" };
        n2.server.updatesEventEmitter.emit("closed");
        expect(n2.status.calledOnceWithExactly({ fill: "red", shape: "ring", text: "disconnected" })).toBeTruthy();
        done();
      });
  });
});