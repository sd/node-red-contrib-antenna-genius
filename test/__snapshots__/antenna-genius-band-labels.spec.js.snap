// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`antenna-genius-band-labels Node should match snapshot 1`] = `
AntennaGeniusBandLabels {
  "_closeCallbacks": Array [],
  "_inputCallback": null,
  "_inputCallbacks": null,
  "_wire": "n3",
  "_wireCount": 1,
  "bandNameA": "",
  "bandNameB": "",
  "g": undefined,
  "id": "n2",
  "name": "test name",
  "send": [Function],
  "server": AntennaGeniusServerNode {
    "_closeCallbacks": Array [
      [Function],
    ],
    "_inputCallback": null,
    "_inputCallbacks": null,
    "_wireCount": 0,
    "activeColor": "Green",
    "antennas": Array [],
    "autoConnect": false,
    "bands": Array [],
    "client": Socket {
      "_events": Object {
        "close": [Function],
        "connect": [Function],
        "end": [Function],
        "error": [Function],
      },
      "_eventsCount": 4,
      "_hadError": false,
      "_host": null,
      "_maxListeners": undefined,
      "_parent": null,
      "_pendingData": null,
      "_pendingEncoding": "",
      "_readableState": ReadableState {
        "autoDestroy": true,
        "awaitDrainWriters": null,
        "buffer": BufferList {
          "head": null,
          "length": 0,
          "tail": null,
        },
        "closeEmitted": false,
        "closed": false,
        "constructed": true,
        "dataEmitted": false,
        "decoder": null,
        "defaultEncoding": "utf8",
        "destroyed": false,
        "emitClose": false,
        "emittedReadable": false,
        "encoding": null,
        "endEmitted": false,
        "ended": false,
        "errorEmitted": false,
        "errored": null,
        "flowing": null,
        "highWaterMark": 16384,
        "length": 0,
        "multiAwaitDrain": false,
        "needReadable": false,
        "objectMode": false,
        "pipes": Array [],
        "readableListening": false,
        "reading": false,
        "readingMore": false,
        "resumeScheduled": false,
        "sync": true,
        Symbol(kPaused): null,
      },
      "_server": null,
      "_sockname": null,
      "_writableState": WritableState {
        "afterWriteTickInfo": null,
        "allBuffers": true,
        "allNoop": true,
        "autoDestroy": true,
        "bufferProcessing": false,
        "buffered": Array [],
        "bufferedIndex": 0,
        "closeEmitted": false,
        "closed": false,
        "constructed": true,
        "corked": 0,
        "decodeStrings": false,
        "defaultEncoding": "utf8",
        "destroyed": false,
        "emitClose": false,
        "ended": false,
        "ending": false,
        "errorEmitted": false,
        "errored": null,
        "finalCalled": false,
        "finished": false,
        "highWaterMark": 16384,
        "length": 0,
        "needDrain": false,
        "objectMode": false,
        "onwrite": [Function],
        "pendingcb": 0,
        "prefinished": false,
        "sync": true,
        "writecb": null,
        "writelen": 0,
        "writing": false,
        Symbol(kOnFinished): Array [],
      },
      "allowHalfOpen": false,
      "connecting": false,
      "server": null,
      Symbol(async_id_symbol): -1,
      Symbol(kHandle): null,
      Symbol(kSetNoDelay): false,
      Symbol(lastWriteQueueSize): 0,
      Symbol(timeout): null,
      Symbol(kBuffer): null,
      Symbol(kBufferCb): null,
      Symbol(kBufferGen): null,
      Symbol(kCapture): false,
      Symbol(kBytesRead): 0,
      Symbol(kBytesWritten): 0,
    },
    "connected": false,
    "disabledColor": "Black",
    "g": undefined,
    "host": undefined,
    "id": "n1",
    "info": Object {},
    "interval": null,
    "name": "test name",
    "port": 9007,
    "selectedColor": "Blue",
    "send": [Function],
    "status": Object {},
    "timer": null,
    "type": "antenna-genius-server",
    "updatesEventEmitter": UpdatesEventEmitter {
      "_events": Object {
        "closed": [Function],
        "connected": [Function],
        "status": [Function],
      },
      "_eventsCount": 3,
      "_maxListeners": 0,
      Symbol(kCapture): false,
    },
    "wires": Array [],
    "z": undefined,
  },
  "type": "antenna-genius-band-labels",
  "wires": Array [
    Array [
      "n3",
    ],
  ],
  "z": undefined,
}
`;

exports[`antenna-genius-band-labels Node should not provide connected status 1`] = `
AntennaGeniusBandLabels {
  "_closeCallbacks": Array [],
  "_inputCallback": null,
  "_inputCallbacks": null,
  "_wire": "n3",
  "_wireCount": 1,
  "bandNameA": "",
  "bandNameB": "",
  "g": undefined,
  "id": "n2",
  "name": "test name",
  "send": [Function],
  "server": AntennaGeniusServerNode {
    "_closeCallbacks": Array [
      [Function],
    ],
    "_inputCallback": null,
    "_inputCallbacks": null,
    "_wireCount": 0,
    "activeColor": "Green",
    "antennas": Array [],
    "autoConnect": false,
    "bands": Array [],
    "client": Socket {
      "_events": Object {
        "close": [Function],
        "connect": [Function],
        "end": [Function],
        "error": [Function],
      },
      "_eventsCount": 4,
      "_hadError": false,
      "_host": null,
      "_maxListeners": undefined,
      "_parent": null,
      "_pendingData": null,
      "_pendingEncoding": "",
      "_readableState": ReadableState {
        "autoDestroy": true,
        "awaitDrainWriters": null,
        "buffer": BufferList {
          "head": null,
          "length": 0,
          "tail": null,
        },
        "closeEmitted": false,
        "closed": false,
        "constructed": true,
        "dataEmitted": false,
        "decoder": null,
        "defaultEncoding": "utf8",
        "destroyed": false,
        "emitClose": false,
        "emittedReadable": false,
        "encoding": null,
        "endEmitted": false,
        "ended": false,
        "errorEmitted": false,
        "errored": null,
        "flowing": null,
        "highWaterMark": 16384,
        "length": 0,
        "multiAwaitDrain": false,
        "needReadable": false,
        "objectMode": false,
        "pipes": Array [],
        "readableListening": false,
        "reading": false,
        "readingMore": false,
        "resumeScheduled": false,
        "sync": true,
        Symbol(kPaused): null,
      },
      "_server": null,
      "_sockname": null,
      "_writableState": WritableState {
        "afterWriteTickInfo": null,
        "allBuffers": true,
        "allNoop": true,
        "autoDestroy": true,
        "bufferProcessing": false,
        "buffered": Array [],
        "bufferedIndex": 0,
        "closeEmitted": false,
        "closed": false,
        "constructed": true,
        "corked": 0,
        "decodeStrings": false,
        "defaultEncoding": "utf8",
        "destroyed": false,
        "emitClose": false,
        "ended": false,
        "ending": false,
        "errorEmitted": false,
        "errored": null,
        "finalCalled": false,
        "finished": false,
        "highWaterMark": 16384,
        "length": 0,
        "needDrain": false,
        "objectMode": false,
        "onwrite": [Function],
        "pendingcb": 0,
        "prefinished": false,
        "sync": true,
        "writecb": null,
        "writelen": 0,
        "writing": false,
        Symbol(kOnFinished): Array [],
      },
      "allowHalfOpen": false,
      "connecting": false,
      "server": null,
      Symbol(async_id_symbol): -1,
      Symbol(kHandle): null,
      Symbol(kSetNoDelay): false,
      Symbol(lastWriteQueueSize): 0,
      Symbol(timeout): null,
      Symbol(kBuffer): null,
      Symbol(kBufferCb): null,
      Symbol(kBufferGen): null,
      Symbol(kCapture): false,
      Symbol(kBytesRead): 0,
      Symbol(kBytesWritten): 0,
    },
    "connected": false,
    "disabledColor": "Black",
    "g": undefined,
    "host": undefined,
    "id": "n1",
    "info": Object {},
    "interval": null,
    "name": "test name",
    "port": 9007,
    "selectedColor": "Blue",
    "send": [Function],
    "status": Object {},
    "timer": null,
    "type": "antenna-genius-server",
    "updatesEventEmitter": UpdatesEventEmitter {
      "_events": Object {
        "closed": [Function],
        "connected": [Function],
        "status": [Function],
      },
      "_eventsCount": 3,
      "_maxListeners": 0,
      Symbol(kCapture): false,
    },
    "wires": Array [],
    "z": undefined,
  },
  "type": "antenna-genius-band-labels",
  "wires": Array [
    Array [
      "n3",
    ],
  ],
  "z": undefined,
}
`;

exports[`antenna-genius-band-labels Node should provide connected status 1`] = `
AntennaGeniusBandLabels {
  "_closeCallbacks": Array [],
  "_inputCallback": null,
  "_inputCallbacks": null,
  "_wire": "n3",
  "_wireCount": 1,
  "bandNameA": "",
  "bandNameB": "",
  "g": undefined,
  "id": "n2",
  "name": "test name",
  "send": [Function],
  "server": AntennaGeniusServerNode {
    "_closeCallbacks": Array [
      [Function],
    ],
    "_inputCallback": null,
    "_inputCallbacks": null,
    "_wireCount": 0,
    "activeColor": "Green",
    "antennas": Array [],
    "autoConnect": false,
    "bands": Array [],
    "client": Socket {
      "_events": Object {
        "close": [Function],
        "connect": [Function],
        "end": [Function],
        "error": [Function],
      },
      "_eventsCount": 4,
      "_hadError": false,
      "_host": null,
      "_maxListeners": undefined,
      "_parent": null,
      "_pendingData": null,
      "_pendingEncoding": "",
      "_readableState": ReadableState {
        "autoDestroy": true,
        "awaitDrainWriters": null,
        "buffer": BufferList {
          "head": null,
          "length": 0,
          "tail": null,
        },
        "closeEmitted": false,
        "closed": false,
        "constructed": true,
        "dataEmitted": false,
        "decoder": null,
        "defaultEncoding": "utf8",
        "destroyed": false,
        "emitClose": false,
        "emittedReadable": false,
        "encoding": null,
        "endEmitted": false,
        "ended": false,
        "errorEmitted": false,
        "errored": null,
        "flowing": null,
        "highWaterMark": 16384,
        "length": 0,
        "multiAwaitDrain": false,
        "needReadable": false,
        "objectMode": false,
        "pipes": Array [],
        "readableListening": false,
        "reading": false,
        "readingMore": false,
        "resumeScheduled": false,
        "sync": true,
        Symbol(kPaused): null,
      },
      "_server": null,
      "_sockname": null,
      "_writableState": WritableState {
        "afterWriteTickInfo": null,
        "allBuffers": true,
        "allNoop": true,
        "autoDestroy": true,
        "bufferProcessing": false,
        "buffered": Array [],
        "bufferedIndex": 0,
        "closeEmitted": false,
        "closed": false,
        "constructed": true,
        "corked": 0,
        "decodeStrings": false,
        "defaultEncoding": "utf8",
        "destroyed": false,
        "emitClose": false,
        "ended": false,
        "ending": false,
        "errorEmitted": false,
        "errored": null,
        "finalCalled": false,
        "finished": false,
        "highWaterMark": 16384,
        "length": 0,
        "needDrain": false,
        "objectMode": false,
        "onwrite": [Function],
        "pendingcb": 0,
        "prefinished": false,
        "sync": true,
        "writecb": null,
        "writelen": 0,
        "writing": false,
        Symbol(kOnFinished): Array [],
      },
      "allowHalfOpen": false,
      "connecting": false,
      "server": null,
      Symbol(async_id_symbol): -1,
      Symbol(kHandle): null,
      Symbol(kSetNoDelay): false,
      Symbol(lastWriteQueueSize): 0,
      Symbol(timeout): null,
      Symbol(kBuffer): null,
      Symbol(kBufferCb): null,
      Symbol(kBufferGen): null,
      Symbol(kCapture): false,
      Symbol(kBytesRead): 0,
      Symbol(kBytesWritten): 0,
    },
    "connected": false,
    "disabledColor": "Black",
    "g": undefined,
    "host": undefined,
    "id": "n1",
    "info": Object {
      "name": "Name",
    },
    "interval": null,
    "name": "test name",
    "port": 9007,
    "selectedColor": "Blue",
    "send": [Function],
    "status": Object {},
    "timer": null,
    "type": "antenna-genius-server",
    "updatesEventEmitter": UpdatesEventEmitter {
      "_events": Object {
        "closed": [Function],
        "connected": [Function],
        "status": [Function],
      },
      "_eventsCount": 3,
      "_maxListeners": 0,
      Symbol(kCapture): false,
    },
    "wires": Array [],
    "z": undefined,
  },
  "type": "antenna-genius-band-labels",
  "wires": Array [
    Array [
      "n3",
    ],
  ],
  "z": undefined,
}
`;

exports[`antenna-genius-band-labels Node should provide disconnected status 1`] = `
AntennaGeniusBandLabels {
  "_closeCallbacks": Array [],
  "_inputCallback": null,
  "_inputCallbacks": null,
  "_wire": "n3",
  "_wireCount": 1,
  "bandNameA": "",
  "bandNameB": "",
  "g": undefined,
  "id": "n2",
  "name": "test name",
  "send": [Function],
  "server": AntennaGeniusServerNode {
    "_closeCallbacks": Array [
      [Function],
    ],
    "_inputCallback": null,
    "_inputCallbacks": null,
    "_wireCount": 0,
    "activeColor": "Green",
    "antennas": Array [],
    "autoConnect": false,
    "bands": Array [],
    "client": Socket {
      "_events": Object {
        "close": [Function],
        "connect": [Function],
        "end": [Function],
        "error": [Function],
      },
      "_eventsCount": 4,
      "_hadError": false,
      "_host": null,
      "_maxListeners": undefined,
      "_parent": null,
      "_pendingData": null,
      "_pendingEncoding": "",
      "_readableState": ReadableState {
        "autoDestroy": true,
        "awaitDrainWriters": null,
        "buffer": BufferList {
          "head": null,
          "length": 0,
          "tail": null,
        },
        "closeEmitted": false,
        "closed": false,
        "constructed": true,
        "dataEmitted": false,
        "decoder": null,
        "defaultEncoding": "utf8",
        "destroyed": false,
        "emitClose": false,
        "emittedReadable": false,
        "encoding": null,
        "endEmitted": false,
        "ended": false,
        "errorEmitted": false,
        "errored": null,
        "flowing": null,
        "highWaterMark": 16384,
        "length": 0,
        "multiAwaitDrain": false,
        "needReadable": false,
        "objectMode": false,
        "pipes": Array [],
        "readableListening": false,
        "reading": false,
        "readingMore": false,
        "resumeScheduled": false,
        "sync": true,
        Symbol(kPaused): null,
      },
      "_server": null,
      "_sockname": null,
      "_writableState": WritableState {
        "afterWriteTickInfo": null,
        "allBuffers": true,
        "allNoop": true,
        "autoDestroy": true,
        "bufferProcessing": false,
        "buffered": Array [],
        "bufferedIndex": 0,
        "closeEmitted": false,
        "closed": false,
        "constructed": true,
        "corked": 0,
        "decodeStrings": false,
        "defaultEncoding": "utf8",
        "destroyed": false,
        "emitClose": false,
        "ended": false,
        "ending": false,
        "errorEmitted": false,
        "errored": null,
        "finalCalled": false,
        "finished": false,
        "highWaterMark": 16384,
        "length": 0,
        "needDrain": false,
        "objectMode": false,
        "onwrite": [Function],
        "pendingcb": 0,
        "prefinished": false,
        "sync": true,
        "writecb": null,
        "writelen": 0,
        "writing": false,
        Symbol(kOnFinished): Array [],
      },
      "allowHalfOpen": false,
      "connecting": false,
      "server": null,
      Symbol(async_id_symbol): -1,
      Symbol(kHandle): null,
      Symbol(kSetNoDelay): false,
      Symbol(lastWriteQueueSize): 0,
      Symbol(timeout): null,
      Symbol(kBuffer): null,
      Symbol(kBufferCb): null,
      Symbol(kBufferGen): null,
      Symbol(kCapture): false,
      Symbol(kBytesRead): 0,
      Symbol(kBytesWritten): 0,
    },
    "connected": false,
    "disabledColor": "Black",
    "g": undefined,
    "host": undefined,
    "id": "n1",
    "info": Object {
      "name": "Name",
    },
    "interval": null,
    "name": "test name",
    "port": 9007,
    "selectedColor": "Blue",
    "send": [Function],
    "status": Object {},
    "timer": null,
    "type": "antenna-genius-server",
    "updatesEventEmitter": UpdatesEventEmitter {
      "_events": Object {
        "closed": [Function],
        "connected": [Function],
        "status": [Function],
      },
      "_eventsCount": 3,
      "_maxListeners": 0,
      Symbol(kCapture): false,
    },
    "wires": Array [],
    "z": undefined,
  },
  "type": "antenna-genius-band-labels",
  "wires": Array [
    Array [
      "n3",
    ],
  ],
  "z": undefined,
}
`;

exports[`antenna-genius-band-labels Node should provide no output with no band definitions 1`] = `
AntennaGeniusBandLabels {
  "_closeCallbacks": Array [],
  "_inputCallback": null,
  "_inputCallbacks": null,
  "_wire": "n3",
  "_wireCount": 1,
  "bandNameA": "",
  "bandNameB": "",
  "g": undefined,
  "id": "n2",
  "name": "test name",
  "send": [Function],
  "server": AntennaGeniusServerNode {
    "_closeCallbacks": Array [
      [Function],
    ],
    "_inputCallback": null,
    "_inputCallbacks": null,
    "_wireCount": 0,
    "activeColor": "Green",
    "antennas": Array [],
    "autoConnect": false,
    "bands": Array [],
    "client": Socket {
      "_events": Object {
        "close": [Function],
        "connect": [Function],
        "end": [Function],
        "error": [Function],
      },
      "_eventsCount": 4,
      "_hadError": false,
      "_host": null,
      "_maxListeners": undefined,
      "_parent": null,
      "_pendingData": null,
      "_pendingEncoding": "",
      "_readableState": ReadableState {
        "autoDestroy": true,
        "awaitDrainWriters": null,
        "buffer": BufferList {
          "head": null,
          "length": 0,
          "tail": null,
        },
        "closeEmitted": false,
        "closed": false,
        "constructed": true,
        "dataEmitted": false,
        "decoder": null,
        "defaultEncoding": "utf8",
        "destroyed": false,
        "emitClose": false,
        "emittedReadable": false,
        "encoding": null,
        "endEmitted": false,
        "ended": false,
        "errorEmitted": false,
        "errored": null,
        "flowing": null,
        "highWaterMark": 16384,
        "length": 0,
        "multiAwaitDrain": false,
        "needReadable": false,
        "objectMode": false,
        "pipes": Array [],
        "readableListening": false,
        "reading": false,
        "readingMore": false,
        "resumeScheduled": false,
        "sync": true,
        Symbol(kPaused): null,
      },
      "_server": null,
      "_sockname": null,
      "_writableState": WritableState {
        "afterWriteTickInfo": null,
        "allBuffers": true,
        "allNoop": true,
        "autoDestroy": true,
        "bufferProcessing": false,
        "buffered": Array [],
        "bufferedIndex": 0,
        "closeEmitted": false,
        "closed": false,
        "constructed": true,
        "corked": 0,
        "decodeStrings": false,
        "defaultEncoding": "utf8",
        "destroyed": false,
        "emitClose": false,
        "ended": false,
        "ending": false,
        "errorEmitted": false,
        "errored": null,
        "finalCalled": false,
        "finished": false,
        "highWaterMark": 16384,
        "length": 0,
        "needDrain": false,
        "objectMode": false,
        "onwrite": [Function],
        "pendingcb": 0,
        "prefinished": false,
        "sync": true,
        "writecb": null,
        "writelen": 0,
        "writing": false,
        Symbol(kOnFinished): Array [],
      },
      "allowHalfOpen": false,
      "connecting": false,
      "server": null,
      Symbol(async_id_symbol): -1,
      Symbol(kHandle): null,
      Symbol(kSetNoDelay): false,
      Symbol(lastWriteQueueSize): 0,
      Symbol(timeout): null,
      Symbol(kBuffer): null,
      Symbol(kBufferCb): null,
      Symbol(kBufferGen): null,
      Symbol(kCapture): false,
      Symbol(kBytesRead): 0,
      Symbol(kBytesWritten): 0,
    },
    "connected": false,
    "disabledColor": "Black",
    "g": undefined,
    "host": undefined,
    "id": "n1",
    "info": Object {},
    "interval": null,
    "name": "test name",
    "port": 9007,
    "selectedColor": "Blue",
    "send": [Function],
    "status": Object {
      "portA_band": 0,
      "portB_band": 1,
    },
    "timer": null,
    "type": "antenna-genius-server",
    "updatesEventEmitter": UpdatesEventEmitter {
      "_events": Object {
        "closed": [Function],
        "connected": [Function],
        "status": [Function],
      },
      "_eventsCount": 3,
      "_maxListeners": 0,
      Symbol(kCapture): false,
    },
    "wires": Array [],
    "z": undefined,
  },
  "type": "antenna-genius-band-labels",
  "wires": Array [
    Array [
      "n3",
    ],
  ],
  "z": undefined,
}
`;

exports[`antenna-genius-band-labels Node should provide no output with no input 1`] = `
AntennaGeniusBandLabels {
  "_closeCallbacks": Array [],
  "_inputCallback": null,
  "_inputCallbacks": null,
  "_wire": "n3",
  "_wireCount": 1,
  "bandNameA": "",
  "bandNameB": "",
  "g": undefined,
  "id": "n2",
  "name": "test name",
  "send": [Function],
  "server": AntennaGeniusServerNode {
    "_closeCallbacks": Array [
      [Function],
    ],
    "_inputCallback": null,
    "_inputCallbacks": null,
    "_wireCount": 0,
    "activeColor": "Green",
    "antennas": Array [],
    "autoConnect": false,
    "bands": Array [],
    "client": Socket {
      "_events": Object {
        "close": [Function],
        "connect": [Function],
        "end": [Function],
        "error": [Function],
      },
      "_eventsCount": 4,
      "_hadError": false,
      "_host": null,
      "_maxListeners": undefined,
      "_parent": null,
      "_pendingData": null,
      "_pendingEncoding": "",
      "_readableState": ReadableState {
        "autoDestroy": true,
        "awaitDrainWriters": null,
        "buffer": BufferList {
          "head": null,
          "length": 0,
          "tail": null,
        },
        "closeEmitted": false,
        "closed": false,
        "constructed": true,
        "dataEmitted": false,
        "decoder": null,
        "defaultEncoding": "utf8",
        "destroyed": false,
        "emitClose": false,
        "emittedReadable": false,
        "encoding": null,
        "endEmitted": false,
        "ended": false,
        "errorEmitted": false,
        "errored": null,
        "flowing": null,
        "highWaterMark": 16384,
        "length": 0,
        "multiAwaitDrain": false,
        "needReadable": false,
        "objectMode": false,
        "pipes": Array [],
        "readableListening": false,
        "reading": false,
        "readingMore": false,
        "resumeScheduled": false,
        "sync": true,
        Symbol(kPaused): null,
      },
      "_server": null,
      "_sockname": null,
      "_writableState": WritableState {
        "afterWriteTickInfo": null,
        "allBuffers": true,
        "allNoop": true,
        "autoDestroy": true,
        "bufferProcessing": false,
        "buffered": Array [],
        "bufferedIndex": 0,
        "closeEmitted": false,
        "closed": false,
        "constructed": true,
        "corked": 0,
        "decodeStrings": false,
        "defaultEncoding": "utf8",
        "destroyed": false,
        "emitClose": false,
        "ended": false,
        "ending": false,
        "errorEmitted": false,
        "errored": null,
        "finalCalled": false,
        "finished": false,
        "highWaterMark": 16384,
        "length": 0,
        "needDrain": false,
        "objectMode": false,
        "onwrite": [Function],
        "pendingcb": 0,
        "prefinished": false,
        "sync": true,
        "writecb": null,
        "writelen": 0,
        "writing": false,
        Symbol(kOnFinished): Array [],
      },
      "allowHalfOpen": false,
      "connecting": false,
      "server": null,
      Symbol(async_id_symbol): -1,
      Symbol(kHandle): null,
      Symbol(kSetNoDelay): false,
      Symbol(lastWriteQueueSize): 0,
      Symbol(timeout): null,
      Symbol(kBuffer): null,
      Symbol(kBufferCb): null,
      Symbol(kBufferGen): null,
      Symbol(kCapture): false,
      Symbol(kBytesRead): 0,
      Symbol(kBytesWritten): 0,
    },
    "connected": false,
    "disabledColor": "Black",
    "g": undefined,
    "host": undefined,
    "id": "n1",
    "info": Object {},
    "interval": null,
    "name": "test name",
    "port": 9007,
    "selectedColor": "Blue",
    "send": [Function],
    "status": Object {},
    "timer": null,
    "type": "antenna-genius-server",
    "updatesEventEmitter": UpdatesEventEmitter {
      "_events": Object {
        "closed": [Function],
        "connected": [Function],
        "status": [Function],
      },
      "_eventsCount": 3,
      "_maxListeners": 0,
      Symbol(kCapture): false,
    },
    "wires": Array [],
    "z": undefined,
  },
  "type": "antenna-genius-band-labels",
  "wires": Array [
    Array [
      "n3",
    ],
  ],
  "z": undefined,
}
`;

exports[`antenna-genius-band-labels Node should provide output 1`] = `
AntennaGeniusBandLabels {
  "_closeCallbacks": Array [],
  "_inputCallback": null,
  "_inputCallbacks": null,
  "_wire": "n3",
  "_wireCount": 1,
  "bandNameA": "bandA",
  "bandNameB": "bandB",
  "g": undefined,
  "id": "n2",
  "name": "test name",
  "send": [Function],
  "server": AntennaGeniusServerNode {
    "_closeCallbacks": Array [
      [Function],
    ],
    "_inputCallback": null,
    "_inputCallbacks": null,
    "_wireCount": 0,
    "activeColor": "Green",
    "antennas": Array [],
    "autoConnect": false,
    "bands": Array [
      Object {
        "band_name": "bandA",
      },
      Object {
        "band_name": "bandB",
      },
    ],
    "client": Socket {
      "_events": Object {
        "close": [Function],
        "connect": [Function],
        "end": [Function],
        "error": [Function],
      },
      "_eventsCount": 4,
      "_hadError": false,
      "_host": null,
      "_maxListeners": undefined,
      "_parent": null,
      "_pendingData": null,
      "_pendingEncoding": "",
      "_readableState": ReadableState {
        "autoDestroy": true,
        "awaitDrainWriters": null,
        "buffer": BufferList {
          "head": null,
          "length": 0,
          "tail": null,
        },
        "closeEmitted": false,
        "closed": false,
        "constructed": true,
        "dataEmitted": false,
        "decoder": null,
        "defaultEncoding": "utf8",
        "destroyed": false,
        "emitClose": false,
        "emittedReadable": false,
        "encoding": null,
        "endEmitted": false,
        "ended": false,
        "errorEmitted": false,
        "errored": null,
        "flowing": null,
        "highWaterMark": 16384,
        "length": 0,
        "multiAwaitDrain": false,
        "needReadable": false,
        "objectMode": false,
        "pipes": Array [],
        "readableListening": false,
        "reading": false,
        "readingMore": false,
        "resumeScheduled": false,
        "sync": true,
        Symbol(kPaused): null,
      },
      "_server": null,
      "_sockname": null,
      "_writableState": WritableState {
        "afterWriteTickInfo": null,
        "allBuffers": true,
        "allNoop": true,
        "autoDestroy": true,
        "bufferProcessing": false,
        "buffered": Array [],
        "bufferedIndex": 0,
        "closeEmitted": false,
        "closed": false,
        "constructed": true,
        "corked": 0,
        "decodeStrings": false,
        "defaultEncoding": "utf8",
        "destroyed": false,
        "emitClose": false,
        "ended": false,
        "ending": false,
        "errorEmitted": false,
        "errored": null,
        "finalCalled": false,
        "finished": false,
        "highWaterMark": 16384,
        "length": 0,
        "needDrain": false,
        "objectMode": false,
        "onwrite": [Function],
        "pendingcb": 0,
        "prefinished": false,
        "sync": true,
        "writecb": null,
        "writelen": 0,
        "writing": false,
        Symbol(kOnFinished): Array [],
      },
      "allowHalfOpen": false,
      "connecting": false,
      "server": null,
      Symbol(async_id_symbol): -1,
      Symbol(kHandle): null,
      Symbol(kSetNoDelay): false,
      Symbol(lastWriteQueueSize): 0,
      Symbol(timeout): null,
      Symbol(kBuffer): null,
      Symbol(kBufferCb): null,
      Symbol(kBufferGen): null,
      Symbol(kCapture): false,
      Symbol(kBytesRead): 0,
      Symbol(kBytesWritten): 0,
    },
    "connected": false,
    "disabledColor": "Black",
    "g": undefined,
    "host": undefined,
    "id": "n1",
    "info": Object {
      "name": "Name",
    },
    "interval": null,
    "name": "test name",
    "port": 9007,
    "selectedColor": "Blue",
    "send": [Function],
    "status": Object {
      "portA_band": 0,
      "portB_band": 1,
    },
    "timer": null,
    "type": "antenna-genius-server",
    "updatesEventEmitter": UpdatesEventEmitter {
      "_events": Object {
        "closed": [Function],
        "connected": [Function],
        "status": [Function],
      },
      "_eventsCount": 3,
      "_maxListeners": 0,
      Symbol(kCapture): false,
    },
    "wires": Array [],
    "z": undefined,
  },
  "type": "antenna-genius-band-labels",
  "wires": Array [
    Array [
      "n3",
    ],
  ],
  "z": undefined,
}
`;
